nomenclature/processor/required_data.py

```diff
@@ -41,14 +42,11 @@ def cast_to_RequiredMeasurand(v) -> RequiredMeasurand:
     return RequiredMeasurand(variable=variable, **v[variable])
 
 
-class RequiredData(BaseModel):
+class RequiredData(IamcDataFilter):
     measurand: (
         List[Annotated[RequiredMeasurand, BeforeValidator(cast_to_RequiredMeasurand)]]
         | None
     ) = None
-    variable: List[str] | None = None
```

Comment:
By changing the base class of `RequiredData` you're introducing some unwanted changes. 
With your changes `RequiredData` now has the additional attributes `model` and `scenario` which it did not have before.
Now the `RequiredDataValidator` which uses `RequiredData` features the `model` attribute twice. 
Once as `RequiredDataValidator.model` and as `RequiredDataValidator.required_data[0].model`. This is an issue since the latter `model` attribute does not do anything. 
As it stands a required data yaml file like this:

```yaml
required_data:
  - model: model_a
  - scenario: scenario_a
  - measurand:
    - Primary Energy:
        unit: [GWh/yr, Mtoe]
    year: [2005, 2010, 2015]
```

would be parsed without any problems **but** the requirement would be applied to **all** models and scenarios rather than the ones written since the rest of `RequiredData` and `RequiredDataValidator` is not equipped to deal with `model` and `scenario`.
If we wanted to actually apply required data only to a specific model, we'd have to do it this way:

```yaml
model: model_a
required_data:
  - measurand:
    - Primary Energy:
        unit: [GWh/yr, Mtoe]
    year: [2005, 2010, 2015]
```

I'd see a number of different solutions to the issue:

1. Remove `model` and `scenario` from `IamcDataFilter` and add them at the `DataValidator(IamcDataFilter)` level
2. Change the way that `RequiredDataValidator` works so that the first yaml example above works. 

My preference would be for option 1

EDIT: I just realized, the same argument as above also applies for `IamcDataFilter.unit`